### PR TITLE
fix: renew Entry object when add to LRU

### PR DESCRIPTION
**Pull request by wanrui**
use lru to add some entry,the e entry point to same memory.

### CODE REVIEW

**Comment by a8m** [comment]
Hey @wanrui and thanks for the contribution. 
Can you elaborate on why this copy is needed (where it fails) and what problem it solves? Thanks

**Comment by wanrui** [comment]
`mc.buf.takeBuffer(pktLen + 4)` get buffers  as Singleton Pattern ， `mc.buf[95]` address is  same to Add LRU e.values[0][0].  SO when mc.buf change lead to lru change. we get an error like this `converting driver.Value type []uint8 (\"@\") to a int64: invalid syntax`， because LRU's values point to buffers  (mysqlConn  buffer)

```
func (mc *mysqlConn) writeCommandPacketStr(command byte, arg string) error {
	// Reset Packet Sequence
	mc.sequence = 0

	pktLen := 1 + len(arg)
	data, err := mc.buf.takeBuffer(pktLen + 4)
	if err != nil {
		// cannot take the buffer. Something must be wrong with the connection
		errLog.Print(err)
		return errBadConnNoWrite
	}

	// Add command byte
	data[4] = command

	// Add arg
	copy(data[5:], arg)


	// Send CMD packet
	return mc.writePacket(data)
}
```

follow is debug image:


image

image

**Comment by wanrui** [comment]
> Hey @wanrui and thanks for the contribution. Can you elaborate on why this copy is needed (where it fails) and what problem it solves? Thanks

@a8m

File: level.go [hunk 1]
```diff
@@ -93,10 +93,19 @@ func NewLRU(maxEntries int) *LRU {
...
 func (l *LRU) Add(_ context.Context, k Key, e *Entry, ttl time.Duration) error {
+	oldBytes, marshalErr:= e.MarshalBinary()
+	if marshalErr!= nil{
+		return marshalErr
+	}
+	newEntry := &Entry{}
+	unmarshalErr := newEntry.UnmarshalBinary(oldBytes)
+	if unmarshalErr != nil {
+		return unmarshalErr
+	}
```
Comment (a8m) [inline level.go:104]:
```suggestion
	buf, err := e.MarshalBinary()
	if err != nil {
		return err
	}
	newEntry := &Entry{}
	if  err := newEntry.UnmarshalBinary(buf); err != nil {
		return err
	}
```

Looks good, just minor style suggestions.

**Comment by wanrui** [comment]
@a8m how to reappear this bug？  In one Function  to Query two data result twice, Use same query builder like follow.
```
query.Count(req.Request.Context())
query.All(req.Request.Context())
```

**Comment by a8m** [comment]
Merged with d8a3fbb547808e76eb006e8f8c26383e49994b58.

**Comment by a8m** [comment]
Thanks a lot for your contribution @wanrui 🙏

**Pull request closed without merging**